libvast/vast/type_set.hpp

```diff
@@ -0,0 +1,31 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/type.hpp"
+
+#include <unordered_set>
+
+namespace vast {
+
+struct type_set {
+  std::unordered_set<vast::type> value;
+
+  template <class Inspector>
+  friend auto inspect(Inspector& f, type_set& x) {
+    return f(caf::meta::type_name("type_set"), x.value);
+  }
+};
```

Comment:
This looks like a one-off at a first glance. Why not just use the `unordered_set` directly?

Reply:
If you just want to return a list of types, I think even a `vector<type>` would be better suited.

Reply:
We must forward-declare it to define a type ID for it. We don't want to include further headers in `fwd.hpp`. This would work as well:

```c++
// fwd.hpp
namespace vast { struct type_set; }
VAST_DEFINE_TYPE_ID((vast::type_set));

// type_set.hpp
struct type_set : std::unordered_set<vast::type> {
  using unordered_set::unordered_set;
};
```

I think `s/std::unordered_set/std::stable_set` for the choice of container.

We use the same pattern as shown here in multiple places in the code base.